File: BackEnd/server/router/RouterTourist.go
package router

import (
	"BackEnd/server/api/tourist"
	"BackEnd/server/use"
	"github.com/gin-gonic/gin"
)

// TouristRouter 游客状态可以访问的接口
func TouristRouter(engine *gin.RouterGroup) {
	// 请求验证码接口
	engine.GET("/captcha", tourist.Captcha)
	// 注册账号接口
	engine.POST("/registerAccount", tourist.RegisterAccount)
	// 登录账号接口
	engine.POST("/loginAccount", tourist.LoginAccount)
	// 请求邮件验证码
	engine.GET("/registerAccountMail", use.MajorIPSleep(), tourist.RegisterAccountMail)
	// 请求文章内容--接口请求的结果 ,设置接口缓存时间为30分钟
	engine.GET("/article/:ID", tourist.GetArticleContent)
	// 获取8个随机推荐论坛
	engine.GET("/RandomRecommendForum", tourist.RandomRecommendForum)
	// 请求首页随机文章
	engine.GET("/ArticleIndex", tourist.ArticleRandomIO)
}

File: web/src/assets/js/RouterTourist.js
import server from "@/assets/js/server";
import {Toast} from "vant";
import expen from "@/assets/js/expen";
import {marked} from "marked"

const TouristApi = {
    // 请求图形验证码
    GetCaptcha: async () => {
        let data
        await server.get("/api/tourist/captcha").then(function (response) {
            data = response.data
        })
        console.log("请求验证码的值", data)
        return data.data
    },
    // 登录账号
    PostAccountLogin: async (account) => {
        let Bool = false
        await server.post("/api/tourist/loginAccount", {
            User: account.UserMail,
            Password: account.PassWord,
            Captcha: account.Captcha
        }).then(function (response) {
            if (response.data.code === 200) {
                // 同时写入本地缓存
                expen.InterFaceLogin(response.data.data)
                Bool = true
            }
        })
        return !Bool
    },
    // 注册账号
    PostRegisterAccount: async (account) => {
        let Bool = false
        if (account.PassWord !== account.PassWordS) {
            Toast("两次密码不一样,请自行检测")
            return !Bool
        }
        await server.post("/api/tourist/registerAccount", {
            User: account.UserMail,
            UserName: account.UserName,
            Password: account.PassWord,
            Captcha: account.MailCaptcha
        }).then(function (response) {
            if (response.data.code === 200) {
                // 同时写入本地缓存
                expen.InterFaceLogin(response.data.data)
                Bool = true
            }
        })
        return !Bool
    },
    // 发送邮件验证码
    PostRegisterAccountMail: async (mail) => {
        await server.get("/api/tourist/registerAccountMail?mail=" + mail)
    },
    // 请求文章内容
    GetArticleContent: async (id) => {
        let data
        await server.get("/api/tourist/article/" + id).then(function (response) {
            data = response.data.data
            console.log("文章返回的内容：",data)
            data.Article.Content = marked(data.Article.Content)
        })
        return data
    },
}

export default TouristApi